lib/zip_source_buffer.c

```diff
@@ -429,30 +429,32 @@ static bool
 buffer_grow_fragments(buffer_t *buffer, zip_uint64_t capacity, zip_error_t *error) {
     zip_buffer_fragment_t *fragments;
     zip_uint64_t *offsets;
+    int result;
+    zip_uint64_t tmp_fragments_capacity, tmp_offsets_capacity, additional_elements;
 
-    if (capacity < buffer->fragments_capacity) {
+    if (capacity <= buffer->fragments_capacity) {
```

Comment:
This should be fine, IMO, requested capacity is still there if it's equal to the already existing one.

Reply:
Yes, that was a bug, your version is correct.